src/components/organisms/AssetActions/Edit/EditComputeDataset.tsx

```diff
@@ -0,0 +1,249 @@
+import { useOcean } from '@oceanprotocol/react'
+import { Formik } from 'formik'
+import React, { ReactElement, useEffect, useState } from 'react'
+import {
+  validationSchema,
+  getInitialValues
+} from '../../../../models/FormEditComputeDataset'
+import { useAsset } from '../../../../providers/Asset'
+import { useUserPreferences } from '../../../../providers/UserPreferences'
+import MetadataPreview from '../../../molecules/MetadataPreview'
+import Debug from './Debug'
+import Web3Feedback from '../../../molecules/Wallet/Feedback'
+import FormEditComputeDataset from './FormEditComputeDataset'
+import styles from './index.module.css'
+import {
+  Logger,
+  DDO,
+  ServiceComputePrivacy,
+  publisherTrustedAlgorithm as PublisherTrustedAlgorithm
+} from '@oceanprotocol/lib'
+import MetadataFeedback from '../../../molecules/MetadataFeedback'
+import { graphql, useStaticQuery } from 'gatsby'
+import axios from 'axios'
+import { queryMetadata, getAssetsNames } from '../../../../utils/aquarius'
+import Loader from '../../../atoms/Loader'
+import web3 from 'web3'
+
+const contentQuery = graphql`
+  query EditComputeDataQuery {
+    content: allFile(
+      filter: { relativePath: { eq: "pages/editComputeDataset.json" } }
+    ) {
+      edges {
+        node {
+          childPagesJson {
+            description
+            form {
+              success
+              successAction
+              error
+              data {
+                name
+                placeholder
+                label
+                help
+                type
+                required
+                sortOptions
+                options
+                multiple
+                rows
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export default function EditComputeDataset({
+  setShowEdit
+}: {
+  setShowEdit: (show: boolean) => void
+}): ReactElement {
+  const data = useStaticQuery(contentQuery)
+  const content = data.content.edges[0].node.childPagesJson
+
+  const { debug } = useUserPreferences()
+  const { ocean, accountId, config } = useOcean()
+  const { metadata, ddo, refreshDdo } = useAsset()
+  const [success, setSuccess] = useState<string>()
+  const [error, setError] = useState<string>()
+  const [algorithms, setAlgorithms] = useState<AlgorithmOption[]>()
+  const query = {
+    page: 1,
+    query: {
+      nativeSearch: 1,
+      query_string: {
+        query: `(service.attributes.main.type:algorithm) -isInPurgatory:true`
+      }
+    },
+    sort: { created: -1 }
+  }
+  const source = axios.CancelToken.source()
+
+  const hasFeedback = error || success
+
+  interface AlgorithmOption {
+    did: string
+    name: string
+  }
+
+  async function getAlgorithms() {
```

Comment:
would suggest moving this method out of this component, either above the component function in this file or as some util method, maybe in `utils/aquarius`. Seems to just need `metadataCacheUri` as param then.

Same goes for the TypeScript interface, this should never be within a component function but always outside.

Reply:
right, I'll move it to `utils/aquarius`, than it could also be used in #343